### ISSUE TITLE
[Issue]: reducing the bank count on an n2elite is buggy

### ISSUE THREAD

**Issue by random11x** [HiddenRamblings/TagMo#708]
### Issue Description

<details>
<summary>Resolved</summary>
When I reduce the bank count the App seems to successfully write the change but the UI still shows the extra banks as populated. Even rescanning the tag does not correct the issue. You have to force quit the app and then rescan it to see the change.
</details>

> #470: When I have random serial on and I choose "write to banks" the file selector pops up and doesnt let you choose anything. As soon as you go to choose a file it pops up with a dialog asking if you want to write the selected file (even though there are none selected) and then if you say yes it writes blank data to each bank.

> #674: i assume this fix would be live... however i am still having this issue. I enable the randomize on clone option and all amiibo still have the same serial.

### Logcat Output

```shell
TagMo 3.9.9 (Play Release) #606836fa
samsung P (9) - 3.00 GB RAM

--------- beginning of main
06-29 04:58:23.912 24331 24472 W AudioCapabilities: Unsupported mime audio/ac4
06-29 04:58:23.912 24331 24472 W AudioCapabilities: Unsupported mime audio/x-ima
06-29 04:58:23.921 24331 24472 W AudioCapabilities: Unsupported mime audio/eac3-joc
06-29 04:58:23.926 24331 24472 W AudioCapabilities: Unsupported mime audio/evrc
06-29 04:58:23.932 24331 24472 W AudioCapabilities: Unsupported mime audio/mpeg-L1
06-29 04:58:23.932 24331 24472 W AudioCapabilities: Unsupported mime audio/mpeg-L2
06-29 04:58:23.933 24331 24472 W AudioCapabilities: Unsupported mime audio/qcelp
06-29 04:58:23.935 24331 24472 W AudioCapabilities: Unsupported mime audio/x-ms-wma
06-29 04:58:23.938 24331 24472 W AudioCapabilities: Unsupported mime audio/evrc
06-29 04:58:23.939 24331 24472 W AudioCapabilities: Unsupported mime audio/qcelp
06-29 04:58:23.949 24331 24472 W VideoCapabilities: Unrecognized profile 4 for video/hevc
06-29 04:58:23.950 24331 24472 W VideoCapabilities: Unsupported mime video/mp43
06-29 04:58:23.953 24331 24472 W VideoCapabilities: Unrecognized profile/level 1/32 for video/mp4v-es
06-29 04:58:23.953 24331 24472 W VideoCapabilities: Unrecognized profile/level 32768/2 for video/mp4v-es
06-29 04:58:23.953 24331 24472 W VideoCapabilities: Unrecognized profile/level 32768/64 for video/mp4v-es
06-29 04:58:23.953 24331 24472 W VideoCapabilities: Unsupported mime video/wvc1
06-29 04:58:23.954 24331 24472 W VideoCapabilities: Unsupported mime video/wvc1
06-29 04:58:23.959 24331 24472 W VideoCapabilities: Unsupported mime video/x-ms-wmv7
06-29 04:58:23.960 24331 24472 W VideoCapabilities: Unsupported mime video/x-ms-wmv8
06-29 04:58:23.970 24331 24472 W cr_VAUtil: Unknown profile: 1 or level: 2 for codec video/avc
06-29 04:58:23.970 24331 24472 W cr_VAUtil: Unknown profile: 65536 or level: 2 for codec video/avc
06-29 04:58:23.970 24331 24472 W cr_VAUtil: Unknown profile: 2 or level: 2 for codec video/avc
06-29 04:58:23.970 24331 24472 W cr_VAUtil: Unknown profile: 8 or level: 2 for codec video/avc
06-29 04:58:23.970 24331 24472 W cr_VAUtil: Unknown profile: 524288 or level: 2 for codec video/avc
```

### Usage Requirements

- [X] I have read and understood the usage requirements for TagMo

**Comment by AbandonedCart** [comment]
The log is automatically copied to the clipboard, as described in the https://github.com/HiddenRamblings/TagMo/wiki#reporting-issues page

**Comment by random11x** [comment]
I have updated the post to have the logcat output. I reproduced the problem and then pasted the output.

**Comment by AbandonedCart** [comment]
No need to clarify. That's the purpose.

**Comment by random11x** [comment]
I downloaded "4.0.0". Unfortunately I still get the same behaviour for all 3 issues.

**Comment by AbandonedCart** [comment]
Sounds like a hardware issue then.

**Comment by CurtisDS** [comment]
With my phone or the n2elite? I have access to another phone I could test. As for the n2Elite, I can write data to it fine using the standard steps:

Scan the n2elite, choose a slot, replace it with a new bin. etc.

I can also edit each bin one by one editing the properties and manually choosing to create a random serial.

So if that works, I'm not sure there would be an issue with the n2elite itself.

I'll go test the other phone and let you know.

**Comment by random11x** [comment]
With my phone or the n2elite? I have access to another phone I could test. As for the n2Elite, I can write data to it fine using the standard steps:

Scan the n2elite, choose a slot, replace it with a new bin. etc.

I can also edit each bin one by one editing the properties and manually choosing to create a random serial.

So if that works, I'm not sure there would be an issue with the n2elite itself.

I'll go test the other phone and let you know.

**Comment by random11x** [comment]
Same exact thing with the other phone.

This was the LogCat info:

```
TagMo 4.0.0 (GitHub Release) #e5906df
Google R (11) - 3.00 GB RAM

--------- beginning of main
07-03 21:49:36.146 10001 10001 E .tagmo.eightbi: Not starting debugger since process cannot load the jdwp agent.
07-03 21:49:36.704 10001 10021 W AdrenoUtils: <ReadGpuID_from_sysfs:197>: Failed to open /sys/class/kgsl/kgsl-3d0/gpu_model
07-03 21:49:36.704 10001 10021 W AdrenoUtils: <ReadGpuID:221>: Failed to read chip ID from gpu_model. Fallback to use the GSL path
07-03 21:49:36.744 10001 10001 W View    : requestLayout() improperly called by android.widget.FrameLayout{7dff23e V.E..V... ......ID 0,0-1440,2418 #2} during layout: running second layout pass
07-03 21:49:36.744 10001 10001 W View    : requestLayout() improperly called by androidx.recyclerview.widget.RecyclerView{1d389f VFED.V... ......ID 0,2428-1440,2428 #7f09010c app:id/foomiibo_list} during layout: running second layout pass
07-03 21:49:36.744 10001 10001 W View    : requestLayout() improperly called by androidx.recyclerview.widget.RecyclerView{dabbdec VFED.V... ......ID 0,0-1440,2414 #7f090081 app:id/browser_content} during layout: running second layout pass
07-03 21:49:36.758 10001 10021 W Gralloc3: mapper 3.x is not supported
```

**Comment by AbandonedCart** [comment]
Seems the issue is the list itself not adjusting the number of rows. It may be related to Android trying to phase out notifydatasetchanged in favor of added, removed, and changed. It wouldn't be the first time they forced everyone to move to a new design and didn't test, complete it, etc.

**Comment by AbandonedCart** [comment]
Are you able to increase the bank count without box issues?

**Comment by CurtisDS** [comment]
> Are you able to increase the bank count without box issues?
On version #da8ff77 it still doesnt update the number of banks shown until you force quit the app and restart it.

**Comment by random11x** [comment]
> Are you able to increase the bank count without box issues?

On version #da8ff77 it still doesn't update the number of banks shown until you force quit the app and restart it.

**Comment by AbandonedCart** [comment]
So increasing doesn't update, either?

**Comment by random11x** [comment]
increasing works. But it always has worked. The bug only happens when reducing the bank count.

**Comment by AbandonedCart** [comment]
There weren't any changes for that in the latest build.

**Comment by random11x** [comment]
oh...well this is the first Ive had it fully crash when shrinking the bank count. So I figured I would give an update

**Comment by AbandonedCart** [comment]
Somehow the reduced list became just the items that were being removed, which meant the number of items was actually how many need to be taken away from the list instead of how many should be left.

TL;DR the values ended up backwards, but it should be fixed now.
